jgrapht-core/src/main/java/org/jgrapht/alg/shortestpath/BhandariKDisjointShortestPaths.java

```diff
@@ -160,124 +72,19 @@ private void prepare(List<E> previousPath)
         for (E originalEdge : previousPath) {
             source = workingGraph.getEdgeSource(originalEdge);
             target = workingGraph.getEdgeTarget(originalEdge);
+            double originalEdgeWeight = workingGraph.getEdgeWeight(originalEdge);
             workingGraph.removeEdge(originalEdge);
             reversedEdge = workingGraph.addEdge(target, source);
             if (reversedEdge != null) {
```

Comment:
Just curious...in Suurballe we always set the reversed edge weight to zero (regardless of whether the reverse edge was already part of the graph) but here we only do it if no reverse edge was pre-existing.  Is that correct?

Reply:
No, it's not. We should always replace two oppositely directed edges by a single edge directed towards the source vertex. Unfortunately, none of the tests covered it. I tried to add some test cases with no success (test passed anyway). If you can think of a test case covering it let me know and I'll add it.